apps/src/templates/manageStudents/manageStudentsClient.js

```diff
@@ -0,0 +1,137 @@
+import {gql, makeVar, useMutation, useReactiveVar} from '@apollo/client';
+import {pick} from 'lodash';
+import {makeEnum} from '@cdo/apps/utils';
+
+export const StudentInputField = makeEnum('name', 'age', 'gender');
+
+export const manageStudentsTypePolicies = {
+  User: {
+    fields: {
+      isEditing: {
+        read(_, {readField}) {
+          const studentId = readField('id');
+          return !!editingStudentInputsVar()[studentId];
+        }
+      }
+    }
+  },
+  // it turns out that if we query for data that is already in our cache using
+  // a different query from the one that put it there, Apollo doesn't know that
+  // it's there and so goes straight to the server. e.g. querying for section
+  // by id then subsequently asking for one of the students in the section
+  // by querying for user by id results in a network request. the type policy below
+  // tells Apollo that when it sees a `user` field in a query, it should attempt
+  // to return a reference from the cache before hitting the network.
+  //
+  // i find it a little strange that this isn't the default behavior. however,
+  // from what i've read about query batching, it sounds like the correct way to
+  // do a "container" query followed by smaller queries in leaf components
+  // (e.g. whole section followed by individual students) is to export the smaller
+  // queries as fragments and compose the container query out of those. so if we
+  // were doing that, the smaller queries would be identical to portions of the
+  // container query and would therefore know to look in the cache.
+  //
+  // FWIW this functionality is no longer needed by my implementation, but i'm
+  // leaving it here as a record of my learnings.
+  //
+  // -CF
+  Query: {
+    fields: {
+      user(_, {args, toReference}) {
+        return toReference({
+          __typename: 'User',
+          id: args.id
+        });
+      }
+    }
+  }
+};
```

Comment:
Highlighting this.  Looks like there are at least three options here:
1) Issue a large query (e.g. by section) at the container level and pass down pieces of the response using props.
2) Issue a large query (e.g. by section) at the container level and define a cache policy so that small queries (e.g. by user) in subcomponents will be fulfilled by the cache.
3) Subcomponents define their small queries as query fragments and the containers gather them and send an array of small queries.  (Need to check if graphql-ruby support this.  Some server implementations only support a single query.)